-1

Tested at: fafb66afa01175d17cc1f26c5a0858096243e34c

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-88a0cf/1301/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-02-2300/src/MagneticField/ParametrizedEngine/src/BFit3D.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-02-2300/src/MagneticField/ParametrizedEngine/src/rz_poly.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-02-2300/src/MagneticField/ParametrizedEngine/src/rz_harm_poly.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-02-2300/src/MagneticField/ParametrizedEngine/src/poly2d_base.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-02-2300/src/MagneticField/ParametrizedEngine/src/ParabolicParametrizedMagneticField.cc 
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-02-2300/src/MagneticField/ParametrizedEngine/src/TkBfield.cc:14:30: error: constexpr variable 'fpar1' must be initialized by a constant expression
  constexpr BCylParam<float> fpar1{
                             ^~~~~~
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-02-2300/src/MagneticField/ParametrizedEngine/src/BCyl.h:32:30: note: non-constexpr function 'sqrt' cannot be used in a constant expression
          hb0(0.5 * prm[2] * std::sqrt(1.0 + ap2)),
                             ^
</pre>

--- a/MagneticField/ParametrizedEngine/src/TkBfield.cc
+++ b/MagneticField/ParametrizedEngine/src/TkBfield.cc
@@ -11,19 +11,19 @@ using namespace magfieldparam;
 
 namespace {
 
-  constexpr BCylParam<float> fpar1{
+  const BCylParam<float> fpar1{
       4.90541f, 17.8768f, 2.02355f, 0.0210538f, 0.000321885f, 2.37511f, 0.00326725f, 2.07656f, 1.71879f};  // 2.0T-2G
-  constexpr BCylParam<float> fpar2{
+  const BCylParam<float> fpar2{
       4.41982f, 15.7732f, 3.02621f, 0.0197814f, 0.000515759f, 2.43385f, 0.00584258f, 2.11333f, 1.76079f};  // 3.0T-2G
-  constexpr BCylParam<float> fpar3{
+  const BCylParam<float> fpar3{
       4.30161f, 15.2586f, 3.51926f, 0.0183494f, 0.000606773f, 2.45110f, 0.00709986f, 2.12161f, 1.77038f};  // 3.5T-2G
-  constexpr BCylParam<float> fpar4{
+  const BCylParam<float> fpar4{
       4.24326f, 15.0201f, 3.81492f, 0.0178712f, 0.000656527f, 2.45818f, 0.00778695f, 2.12500f, 1.77436f};  // 3.8T-2G
-  constexpr BCylParam<float> fpar5{
+  const BCylParam<float> fpar5{
       4.21136f, 14.8824f, 4.01683f, 0.0175932f, 0.000695541f, 2.45311f, 0.00813447f, 2.11688f, 1.76076f};  // 4.0T-2G
   std::string const flds[] = {"2_0T", "3_0T", "3_5T", "3_8T", "4_0T"};
   constexpr float flds_f[] = {2.0, 3.0, 3.5, 3.8, 4.0};
-  BCylParam<float> constexpr fpars[]{fpar1, fpar2, fpar3, fpar4, fpar5};
+  BCylParam<float> const fpars[]{fpar1, fpar2, fpar3, fpar4, fpar5};
 
   BCylParam<float> const& findPar(float fld) {
     for (int i = 0; i < 5; ++i) {

